core/src/test/java/org/apache/iceberg/TestTableUpdatePartitionSpec.java

```diff
@@ -90,6 +90,46 @@ public void testCommitUpdatedSpec() {
     assertThat(table.spec().lastAssignedFieldId()).isEqualTo(1002);
   }
 
+  @TestTemplate
+  public void testCommitFromSpec() {
+    table.updateSpec().addField(bucket("id", 8)).commit();
+
+    // Evolve the spec
+    PartitionSpec evolvedSpec =
+        PartitionSpec.builderFor(table.schema())
+            .withSpecId(1)
+            .bucket("data", 16)
+            .bucket("id", 8, "id_bucket_8")
+            .build();
+    assertThat(table.spec())
+        .as("Should append a partition field to the spec")
+        .isEqualTo(evolvedSpec);
+    assertThat(table.spec().lastAssignedFieldId()).isEqualTo(1001);
+
+    // Restart the spec
+    table
+        .updateSpec()
+        .fromSpec(PartitionSpec.builderFor(table.schema()).build())
```

Comment:
Shouldn't this just be PartitionSpec.unpartitioned()?

Reply:
This won't work as `PartitionSpec.unpartitioned()` has an empty schema, and the line after (`.addField(bucket("data", 16))`) would throw: `Cannot find field 'data' in struct: struct<>`